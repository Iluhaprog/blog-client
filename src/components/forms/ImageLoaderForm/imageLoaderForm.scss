.image-loader-form {
    width: 100%;
    padding: 20px;

    .drop-zone {
        max-width: 250px;
        min-height: 150px;
        width: 100%;
        margin-bottom: 40px;

        
        img.empty {
            margin-top: 55px;
        }
    }
    
    .image-loader-title {
        margin-bottom: 20px;
    }

    .button { 
        max-width: 250px;
        width: 100%;
    }
}