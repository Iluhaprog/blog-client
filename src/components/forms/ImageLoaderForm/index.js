import { reduxForm } from 'redux-form'; 
import ilf from './ImageLoaderForm';

const ImageLoaderForm = reduxForm({
    form: 'imageLoaderForm',
})(ilf);

export {
    ImageLoaderForm,
};