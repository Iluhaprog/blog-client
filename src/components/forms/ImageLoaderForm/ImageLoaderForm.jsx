import React from 'react';
import { Field } from 'redux-form';
import { SendButton } from '../../buttons';
import { DropZone } from '../../formElements';
import './imageLoaderForm.scss';

function ImageLoaderForm(props) {
    const { handleSubmit } = props;

    return (
        <form onSubmit={handleSubmit} className='image-loader-form'>
            <div className='column column_ai-c'>
            <h1 className='image-loader-title'>Images</h1>
                <Field 
                    name='file'
                    component={DropZone}
                />
                <SendButton 
                    text='Ok'
                />
            </div>
        </form>
    );
}

export default ImageLoaderForm;