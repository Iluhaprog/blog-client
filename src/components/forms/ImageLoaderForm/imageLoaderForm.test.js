import React from 'react';
import { reduxForm } from 'redux-form';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ImageLoaderForm from './ImageLoaderForm';

const mockStore = configureMockStore();
const decor = el => {
    const form = () => (el);
    return reduxForm({
        form: 'test',
    })(form);
}

describe('Test view image location form', () => {
    test('Should view form', () => {
        const store = mockStore();
        const ILF = decor(<ImageLoaderForm onSubmit={() => {}}/>);
        const { asFragment } = render(
            <Provider store={store}>
                <ILF />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    })
});

describe('Test events image location form', () => {
    test('Should handle submit event', async () => {
        const store = mockStore();
        const handleSubmit = jest.fn();
        const ILF = decor(<ImageLoaderForm handleSubmit={handleSubmit}/>);
        const { container } = render(
            <Provider store={store}>
                <ILF />
            </Provider>
        );
        fireEvent.submit(container.getElementsByTagName('form')[0]);
        await waitFor(() => {
            expect(handleSubmit).toHaveBeenCalledTimes(1);
        });
    })
});