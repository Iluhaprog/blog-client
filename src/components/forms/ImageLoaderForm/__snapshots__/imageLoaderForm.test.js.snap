// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view image location form Should view form 1`] = `
<DocumentFragment>
  <form
    class="image-loader-form"
  >
    <div
      class="column column_ai-c"
    >
      <h1
        class="image-loader-title"
      >
        Images
      </h1>
      <div
        class="drop-zone "
      >
        <img
          alt="empty"
          class="empty"
          src="picture.svg"
        />
        <input
          type="file"
        />
      </div>
      <button
        class="button send-button"
      >
        Ok
      </button>
    </div>
  </form>
</DocumentFragment>
`;
